src/slurm_plugin/clustermgtd.py

```diff
@@ -440,47 +438,28 @@ def manage_cluster(self):
-                log.debug("Current inactive slurm nodes in scheduler: %s", inactive_nodes)
-                self._inactive_partitions = get_inactive_partition_names(partitions)
-            except ClusterManager.SchedulerUnavailable:
-                log.error("Unable to get partition/node info from slurm, no other action can be performed. Sleeping...")
-                return
+            active_nodes, inactive_nodes, partition_names_map = self._initialize_nodes_and_partitions()
+
             # Get all non-terminating instances in EC2
-            try:
-                # After reading Slurm nodes wait for 5 seconds to let instances appear in EC2 describe_instances call
-                time.sleep(5)
-                log.info("Retrieving list of EC2 instances associated with the cluster")
-                cluster_instances = self._get_ec2_instances()
-            except ClusterManager.EC2InstancesInfoUnavailable:
-                log.error("Unable to get instances info from EC2, no other action can be performed. Sleeping...")
-                return
-            # Initialize mapping data structures
-            ip_to_slurm_node_map = {node.nodeaddr: node for node in active_nodes}
-            log.debug("ip_to_slurm_node_map %s", ip_to_slurm_node_map)
+            cluster_instances = self._get_cluster_instances()
+            log.debug("Current cluster instances in EC2: %s", cluster_instances)
+
+            # Initialize nodes
+            instance_id_to_active_node_map, inactive_instance_ids, orphaned_instances = self._map_instances_and_nodes(
+                active_nodes, inactive_nodes, cluster_instances
+            )
             # Handle inactive partition and terminate backing instances
             if inactive_nodes:
-                cluster_instances = self._clean_up_inactive_partition(inactive_nodes, cluster_instances)
-            log.debug("Current cluster instances in EC2: %s", cluster_instances)
-            private_ip_to_instance_map = {instance.private_ip: instance for instance in cluster_instances}
-            self._instance_ips_in_cluster = set(private_ip_to_instance_map.keys())
-            log.debug("private_ip_to_instance_map %s", private_ip_to_instance_map)
+                self._clean_up_inactive_partition(inactive_nodes, inactive_instance_ids)
             # Clean up orphaned instances and skip all other operations if no active node
             if active_nodes:
                 # Perform health check actions
                 if not self._config.disable_all_health_checks:
-                    self._perform_health_check_actions(cluster_instances, ip_to_slurm_node_map)
+                    self._perform_health_check_actions(instance_id_to_active_node_map)
                 # Maintain slurm nodes
-                self._maintain_nodes(private_ip_to_instance_map, active_nodes, partitions)
+                self._maintain_nodes(active_nodes, partition_names_map)
```

Comment:
Why do we need a `partition_names_map`? I would think a list of `SlurmPartition` object should be sufficient for most functionalities?